x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -75,6 +77,47 @@ export const signalRulesAlertType = ({
       } = params;
       // TODO: Remove this hard extraction of name once this is fixed: https://github.com/elastic/kibana/issues/50522
       const savedObject = await services.savedObjectsClient.get('alert', alertId);
+      const ruleStatusSavedObjects = await services.savedObjectsClient.find<
+        IRuleSavedAttributesSavedObjectAttributes
+      >({
+        type: ruleStatusSavedObjectType,
+        perPage: 6, // 0th element is current status, 1-5 is last 5 failures.
+        sortField: 'statusDate',
+        sortOrder: 'desc',
+        search: `${alertId}`,
+        searchFields: ['alertId'],
+      });
+      logger.debug(`ruleStatusSavedObjects: ${JSON.stringify(ruleStatusSavedObjects, null, 4)}`);
+      let currentStatusSavedObject;
+      if (ruleStatusSavedObjects.saved_objects.length === 0) {
+        // create
+        const date = new Date().toISOString();
+        currentStatusSavedObject = await services.savedObjectsClient.create<
+          IRuleSavedAttributesSavedObjectAttributes
+        >(ruleStatusSavedObjectType, {
+          alertId: `${alertId}`, // do a search for this id.
+          statusDate: date,
+          status: 'executing',
+          lastFailureAt: '1970-01-01T00:00:00Z', // default to unix epoch time
+          lastSuccessAt: '1970-01-01T00:00:00Z',
+          lastFailureMessage: '',
+          lastSuccessMessage: '',
```

Comment:
I would default any unknowns as `null` in the data store and on return statements the UI might or might not get these status's rather than giving it hard coded sentinel values. The contract for the front end is optionally we return the lastFailureAt, lastSuccessAt,  lastFailureMessage, lastSuccessMessage but if we don't assume they are not set.